docs/examples/word_embedding/model.py

```diff
@@ -0,0 +1,197 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=
+"""Word embedding models."""
+
+import mxnet as mx
+import numpy as np
+
+import gluonnlp as nlp
+
+
+class Net(mx.gluon.HybridBlock):
+    """Base class for word2vec and fastText SkipGram and CBOW networks.
+
+    Parameters
+    ----------
+    token_to_idx : dict
+        token_to_idx mapping of the vocabulary that this model is to be trained
+        with. token_to_idx is used for __getitem__ and __contains__. For
+        len(token_to_idx) is used during initialization to obtain the input_dim
+        of the embedding matrix.
+    output_dim : int
+        Dimension of the dense embedding.
+    batch_size : int
+        Batchsize this model will be trained with. TODO temporary until
```

Comment:
random_like is merged, though it's unclear whether the mxnet release can meet our timeline.